src/main.tsx

```diff
@@ -1,9 +1,20 @@
 import ReactDOM from 'react-dom/client'
 import App from './App.tsx'
 import { BrowserRouter } from 'react-router-dom'
+import { ColoredPokemonsContextProvider } from './context/ColoredPokemonsContext.tsx'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools'
+
+// Create a client
+const queryClient = new QueryClient()
 
 ReactDOM.createRoot(document.getElementById('root')!).render(
-    <BrowserRouter>
-        <App />
-    </BrowserRouter>
+    <ColoredPokemonsContextProvider>
+        <BrowserRouter>
+            <QueryClientProvider client={queryClient}>
+                <App />
+                <ReactQueryDevtools initialIsOpen={false} />
+            </QueryClientProvider>
+        </BrowserRouter>
+    </ColoredPokemonsContextProvider>
```

Comment:
always use [StrictMode](https://react.dev/reference/react/StrictMode) wrapper as it helps find bugs in your components early during development.